packages/serializer/addon/rest.js

```diff
@@ -286,8 +286,10 @@ const RESTSerializer = JSONSerializer.extend({
           'The adapter returned an array for the primary data of a `queryRecord` response. This is deprecated as `queryRecord` should return a single record.';
 
         deprecate(message, !isQueryRecordAnArray, {
-          id: 'ds.serializer.rest.queryRecord-array-response',
```

Comment:
I have concerns about changing the ID of the deprecation post-release. Also I think I agree that we should just change this to an assert at this point. Thoughts @rwjblue ?

Reply:
Ya, agree.

Reply:
Updated.